README.rst

```diff
@@ -411,44 +411,83 @@ Usual workflow with `Collections`_ would be::
 
 Collections are available on project level only.
 
-Frontier
---------
+Frontiers
+---------
 
 Typical workflow with `Frontier`_::
 
-    >>> frontier = project.frontier
+    >>> frontiers = project.frontiers
 
-Add a request to the frontier::
+Get all frontiers from a project to iterate through it::
 
-    >>> frontier.add('test', 'example.com', [{'fp': '/some/path.html'}])
-    >>> frontier.flush()
-    >>> frontier.newcount
-    1
+    >>> frontiers.iter()
+    <list_iterator at 0x103c93630>
+
+List all frontiers::
+
+    >>> frontiers.list()
+    ['test', 'test1', 'test2']
+
+Get a frontier by name::
+
+    >>> frontier = frontiers.get('test')
+    >>> frontier
+    <scrapinghub.client.Frontier at 0x1048ae4a8>
+
+Get an iterator to iterate through a frontier slots::
+
+    >>> frontier.iter()
+    <list_iterator at 0x1030736d8>
+
+List all slots::
+
+    >>> frontier.list()
+    ['example.com', 'example.com2']
+
+Get a frontier slot by name::
+
+    >>> frontier.get('example.com')
+    <scrapinghub.client.FrontierSlot at 0x1049d8978>
+
+Add a request to the slot::
+
+    >>> slot.add([{'fp': '/some/path.html'}])
+    >>> slot.flush()
 
 Add requests with additional parameters::
 
-    >>> frontier.add('test', 'example.com', [{'fp': '/'}, {'fp': 'page1.html', 'p': 1, 'qdata': {'depth': 1}}])
-    >>> frontier.flush()
-    >>> frontier.newcount
-    2
+    >>> slot.add([{'fp': '/'}, {'fp': 'page1.html', 'p': 1, 'qdata': {'depth': 1}}])
+    >>> slot.flush()
 
-To delete the slot ``example.com`` from the frontier::
+To list requests for a given slot::
 
-    >>> frontier.delete_slot('test', 'example.com')
+    >>> reqs = slot.list()
 
-To retrieve requests for a given slot::
+To retrieve fingerprints for a given slot::
 
-    >>> reqs = frontier.read('test', 'example.com')
+    >>> fps = [req['requests'] for req in slot.iter()]
 
 To delete a batch of requests::
 
-    >>> frontier.delete('test', 'example.com', '00013967d8af7b0001')
+    >>> slot.delete('00013967d8af7b0001')
 
-To retrieve fingerprints for a given slot::
+Flush data of the given frontier::
 
-    >>> fps = [req['requests'] for req in frontier.read('test', 'example.com')]
+    >>> frontier.flush()
+
+To delete the slot ``example.com`` from the frontier::
+
+    >>> frontier.delete('example.com')
```

Comment:
just realized this can be confusing - usually it's expected to call delete on the object that should be deleted. e.g. we have Job.delete() method. What if we move `delete` to FrontierSlot, if no arguments are passed the entier slot is removed, if ids are passed - we remove only some requests?